### PR TITLE
Fix illegal instruction crash on some ARM systems

### PR DESCRIPTION
**Pull request by SChernykh**
See https://github.com/xmrig/xmrig/issues/1916

### CODE REVIEW

**Comment by SChernykh** [comment]
@tevador Can you merge this?

File: src/jit_compiler_a64.cpp [hunk 1]
```diff
@@ -95,6 +95,10 @@ JitCompilerA64::JitCompilerA64()
 {
 	memset(reg_changed_offset, 0, sizeof(reg_changed_offset));
 	memcpy(code, (void*) randomx_program_aarch64, CodeSize);
+
+#ifdef __GNUC__
```
Comment (hyc) [inline src/jit_compiler_a64.cpp:99]: Is this really gcc-specific? The Android SDK uses clang, will this work there, or is some other fix needed for an Android build?

Comment (SChernykh) [reply src/jit_compiler_a64.cpp:99]: It's a GCC-specific builtin, but clang supports it and even defines `__GNUC__`. Also, the same `#ifdef` is used in other places here.

**Comment by Edan3blov** [comment]
Hello,

This code was added and compiled but still mining crashs on smarphones.

If JIT is disabled mining works but without JIT there is no serius mining.

**Pull request merged**